### PR TITLE
solution

### PR DESCRIPTION
**Pull request by igor-kashchenko**

### CODE REVIEW

File: src/controllers/expense.controller.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    if (!isUserIdInvalid) {
+      res.sendStatus(422);
+
+      return;
+    }
+  }
+
+  if (from && to) {
+    const isFromValid = typeof from === 'string';
+    const isToValid = typeof to === 'string';
```
Comment (nzrii-kysel) [inline src/controllers/expense.controller.js:26]: you got those from query, they are always gonna be strings

File: src/controllers/expense.controller.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+    if (!isFromValid || !isToValid) {
+      res.sendStatus(422);
+
+      return;
+    }
+  }
+
+  if (category) {
+    const isCategoryValid = typeof category === 'string';
```
Comment (nzrii-kysel) [inline src/controllers/expense.controller.js:36]: category can be an array too

File: src/controllers/expense.controller.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+  res.send(filteredExpenses);
+};
+
+const getOne = (req, res) => {
+  const expenseId = Number(req.params.expenseId);
+  const isExpenseValid = !isNaN(expenseId);
+
+  if (!isExpenseValid) {
+    res.sendStatus(422);
```
Comment (nzrii-kysel) [inline src/controllers/expense.controller.js:66]:
```suggestion
    res.sendStatus(400);
```

File: src/controllers/expense.controller.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    note,
+  } = req.body;
+
+  const isDataValid = !(
+    !userId
+    || !spentAt
+    || !title
+    || !amount
+    || !category
+  );
```
Comment (nzrii-kysel) [inline src/controllers/expense.controller.js:104]: !(!condition1 || !condition2) === condition1 && condition2

File: src/controllers/expense.controller.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    return;
+  }
+
+  const isDataTypeValid = !(
+    !typeof userId === 'number'
+    || !typeof spentAt === 'string'
+    || !typeof title === 'string'
+    || !typeof amount === 'number'
+    || !typeof category === 'string'
+    || !typeof note === 'string'
```
Comment (nzrii-kysel) [inline src/controllers/expense.controller.js:118]: gonna always be false if request body doesnt have `note` field

File: src/controllers/expense.controller.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+    return;
+  }
+
+  const options = req.body;
+
+  expenseService.update(
+    expenseId,
+    options
+  );
```
Comment (nzrii-kysel) [inline src/controllers/expense.controller.js:207]: forgot validation

**Review by 0xNazarii — CHANGES_REQUESTED** [review]
good job 👍, code needs a bit more work

**Comment by igor-kashchenko** [comment]
> Thanks for your comments. That's all I could fix

**Review by andrii-andriushchenko — APPROVED** [review]
gj 👍
sorry for the dublicate, platform does not want to change to someone's another task 😞

File: src/controllers/expense.controller.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  const expenseId = Number(req.params.expenseId);
+  const isExpenseValid = !isNaN(expenseId);
+
+  if (!isExpenseValid) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!expenseId) {
```
Comment (Anna-Tkachenko) [inline src/controllers/expense.controller.js:23]: can we combine these two checks?

File: src/createServer.js [hunk 8]
```diff
@@ -1,11 +1,24 @@
...
+const { expenseRouter } = require('./routes/expense.router.js');
+const { userService } = require('./services/user.service.js');
+const { expenseService } = require('./services/expense.service.js');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  userService.getInitial();
+  expenseService.getInitial();
```
Comment (Anna-Tkachenko) [inline src/createServer.js:13]: the reason for calling these methods is not clear, can you explain it?

**Review by Anna-Tkachenko — APPROVED** [review]
Good job! ⭐️

**Pull request closed without merging**